src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,166 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final int NUMBER_TO_MULTIPLY = 2;
+    private static final int ZERO = 0;
+    private Node<K, V>[] initialArray;
+    private int size;
+    private int threshold = 12;
+
+    public MyHashMap() {
+        initialArray = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (key == null) {
+            putNullKey(value);
+            return;
+        }
+        Node<K, V> node = newNode(key, value);
+        int index = Math.abs(getHash(key)) % initialArray.length;
+        int loader = threshold;
+        if (initialArray[index] == null) {
+            initialArray[index] = node;
+            size++;
+        } else {
+            Node<K, V> current = initialArray[index];
+            while (current.next != null) {
+                if (Objects.equals(current.key, key)) {
+                    current.value = value;
+                    return;
+                }
+                current = current.next;
+            }
+            if (Objects.equals(current.key, key)) {
+                current.value = value;
+            } else {
+                current.next = node;
+                size++;
+            }
+        }
+        if (size > loader) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        if (key == null) {
```

Comment:
Why logic of handling null key is separated? it is possible to unite it and remove code duplication, you just need to return  bucket index 0 when key is null